addons/project/tests/test_project_flow.py

```diff
@@ -117,3 +120,74 @@ def test_subtask_process(self):
+        Task = self.env['project.task'].with_context({'tracking_disable': True})
+        first_task = Task.create({
+            'name': 'first task',
+            'user_id': self.user_projectuser.id,
+            'project_id': self.project_pigs.id,
+            'partner_id': self.partner_2.id,
+            'planned_hours': 0,
+        })
+
+        self.assertEqual(first_task.rating_count, 0, "Task should have no rating associated with it")
+
+        Rating = self.env['rating.rating']
+        rating_good = Rating.create({
+            'res_model_id': self.env['ir.model']._get('project.task').id,
+            'res_id': first_task.id,
+            'parent_res_model_id': self.env['ir.model']._get('project.project').id,
+            'parent_res_id': self.project_pigs.id,
+            'rated_partner_id': self.partner_2.id,
+            'partner_id': self.partner_2.id,
+            'rating': 10,
+            'consumed': True,
+        })
+
+        rating_bad = Rating.create({
+            'res_model_id': self.env['ir.model']._get('project.task').id,
+            'res_id': first_task.id,
+            'parent_res_model_id': self.env['ir.model']._get('project.project').id,
+            'parent_res_id': self.project_pigs.id,
+            'rated_partner_id': self.partner_2.id,
+            'partner_id': self.partner_2.id,
+            'rating': 5,
+            'consumed': True,
+        })
+
+        self.assertEqual(rating_good.rating_text, 'satisfied')
+        self.assertEqual(rating_bad.rating_text, 'not_satisfied')
+
+        good_img = base64.b64encode(open(get_resource_path('rating', 'static/src/img', 'rating_10.png'), 'rb').read())
+        bad_img = base64.b64encode(open(get_resource_path('rating', 'static/src/img', 'rating_5.png'), 'rb').read())
+
+        self.assertEqual(rating_good.rating_image, good_img)
+        self.assertEqual(rating_bad.rating_image, bad_img)
+
+        # Seems we need this to trigger computation of rating_count
+        first_task._compute_rating_count()
+
+        self.assertEqual(first_task.rating_count, 2, "Task should have two ratings associated with it")
+
+        # Seems we need this to compute the percentage
+        self.project_goats._compute_percentage_satisfaction_task()
+        self.project_pigs._compute_percentage_satisfaction_task()
+
+        self.assertEqual(rating_good.parent_res_id, self.project_pigs.id)
+
+        self.assertEqual(self.project_goats.percentage_satisfaction_task, -1)
+        self.assertEqual(self.project_pigs.percentage_satisfaction_task, 50)
+
+        # We change the task from project_pigs to project_goats, ratings should be associated with the new project
+        first_task.project_id = self.project_goats.id
+
+        # Seems we need this to compute the percentage
+        self.project_goats._compute_percentage_satisfaction_task()
+        self.project_pigs._compute_percentage_satisfaction_task()
```

Comment:
Why should we trigger it manually ? Normally this ORM should trigger it as the depends contains 'tasks.rating_ids.rating'

Reply:
I agree it is a strange behaviour, I don't know why but when I don't force the compute using this method, it doesn't compute anything. I get -1 as the percentage_satisfaction_task. As soon as I put this back it computes it perfectly.